### ISSUE TITLE
Collection expressions lose null safety.

### ISSUE THREAD

**Issue by borland** [dotnet/roslyn#72051]
**Version Used**: 

.NET 8.0.101 SDK

**Steps to Reproduce**:

Observe the following example

```csharp
static List<string?> NullableStrings() => ["c", null, "d"];
List<string> stuff = NullableStrings().ToList();
```

When compiling this code, a compiler warning is correctly raised, because you can't assign `List<string?>` to a `List<string>`.

However, when using a collection expression, no warning or error is raised.

```csharp
List<string> stuff = [..NullableStrings()];
```

[Link to the above example on SharpLab](https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQAwAIpwHQBkCWAdgI4DcyamOAwgPYA29ApgMYAu+thAzuShlgAsfCgGZMMdNXQBvZOgUSY8xXKSKNVAGzoC3NgB4sqAPwA+dADkArowCGAI2YBlNgCciAc24AKAJToALwWANoARCxhADTohLb0MWHAYQC6fJqKKhkA9Nl6hsYW+tYAZiVBVvGOLu5evn7YACq0+f7pGbr4+kZwqEVspeWB6CHY2Db2TkyuHoTe/mlZ6AC+yMtAA===)

As shown in SharpLab, the C# compiler is generating code equivalent to this:

```csharp
List<string> list = new List<string>();
foreach(string? current in NullableStrings())
{
   list.Add(current);
}
stuff = list;
```

If this were hand-written code, a null-safety warning would be emitted on the call to `list.Add`, as it is invalid to put a `string?` into a `List<string>`.
However this warning is not propagated through the compiler/codegen and is not shown to the end user.

**Diagnostic Id**:

n/a

**Expected Behavior**:

A null-safety warning/error should be raised.

**Actual Behavior**:

No warning/error is raised, leading to unexpected possible null values later in the program.

**Comment by CyrusNajmabadi** [comment]
@cston @RikkiGibson this is odd.  I would expect a warning here.

**Comment by cston** [comment]
@borland, thanks for reporting this issue. Yes, nullable analysis of spread elements needs to be added, see https://github.com/dotnet/roslyn/issues/68786.

**Comment by RikkiGibson** [comment]
We fixed this somewhere along the way, I now get a warning in the following scenario:

```cs
static List<string?> NullableStrings() => ["c", null, "d"];
List<string> stuff = [..NullableStrings()]; // warning: Possible null reference argument for parameter 'item' in 'void List<string>.Add(string item)'.
```
